templates/cluster-template.yaml

```diff
@@ -66,7 +66,7 @@ spec:
     files:
     - content: |
         {
-          "cloud": "AzurePublicCloud",
+          "cloud": "${AZURE_ENVIRONMENT}",
```

Comment:
this makes `AZURE_ENVIRONMENT` a required variable, it needs to be added to the CAPI book quickstart

Reply:
is there a different way of doing this so it's optional?

Reply:
Not right now. Checkout https://github.com/kubernetes-sigs/cluster-api/pull/3052

I hesitate to say this, but perhaps, another flavor for the time being?

Reply:
but its not in the templates? how would the flavor do that?

Reply:
I'm not sure I understand the question.

What I'm suggesting is that `cluster-teamplate.yaml` stays as it is with `"AzurePublicCloud"`, so there is no new variable required when deploying the default template. Secondly, create a new flavor which includes the "${AZURE_ENVIRONMENT}"` var.

Does that make sense?

Reply:
got it, ok makes sense, will look into what needs to change

Reply:
Going to play devil's advocate here... I really don't like the idea of having separate flavors for different variable values. I really wish we could do defaulting instead. Since that's not possible, maybe we can just document that you have to change the `cloud` value to the Azure cloud you want to use since this is part of azure.json. Also keep in mind this whole thing will be an environment variable after https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/653/files#diff-1bf07101845de6da98f1c948b82c5ce0R296.

Reply:
it currently defaults to AzurePublicCloud if its empty, or at least it should

Reply:
Are we ok in concept to add a new required variable?

Reply:
Should we just wait until #653 is merged? As @CecileRobertMichon pointed out, once that is merged, there is no required variable :).

Reply:
that's waiting for CAPI 0.3.7 I guess

Reply:
The problem with adding a required var is that it will a breaking change for anyone that will run `clusterctl config cluster --infrastructure=azure`. That will break any scripts using the default flavor.

It would be friendlier to users if we were to leave `"cloud": "AzurePublicCloud"` and provide documentation on how to change the value.

When #653 lands, that documentation is still valid, and we didn't break anyone in the mean time.

Reply:
don't we still have to add AzureEnvironment env var to the capz-controller which would have to come from a secret or similar? I think instructions to update the secret value or change the deployment configuration would not be very user friendly

Reply:
For now let's keep the new var and try to make it optional later with https://github.com/kubernetes-sigs/cluster-api/issues/3189 @nader-ziada